code/game/objects/items/weaponry.dm

```diff
@@ -732,28 +735,66 @@ for further reading, please see: https://github.com/tgstation/tgstation/pull/301
 		target.throw_at(throw_target, rand(8,10), 14, user)
 		SSexplosions.medturf += throw_target
 		playsound(get_turf(src), 'sound/weapons/homerun.ogg', 100, TRUE)
-		homerun_ready = 0
+		homerun_ready = FALSE
 		return
 	else if(!target.anchored)
 		var/whack_speed = (prob(60) ? 1 : 4)
 		target.throw_at(throw_target, rand(1, 2), whack_speed, user, gentle = TRUE) // sorry friends, 7 speed batting caused wounds to absolutely delete whoever you knocked your target into (and said target)
 
+/obj/item/melee/baseball_bat/pre_attack(atom/movable/target, mob/living/user, params)
+	if(!istype(target))
+		return ..()
+	if(!target.throwing || (ismob(target) && !mob_thrower))
+		return ..()
+	var/datum/thrownthing/throw_datum = target.throwing
+	var/datum_throw_speed = throw_datum.speed
+	var/angle = 0
+	var/target_to_user = get_dir(target, user)
+	if(target.dir & turn(target_to_user, 90))
+		angle = 270
+	if(target.dir & turn(target_to_user, 270))
+		angle = 90
+	if(target.dir & turn(target_to_user, 180))
+		angle = 180
+	if(target.dir & target_to_user)
+		angle = 360
+	var/turf/return_to_sender = get_ranged_target_turf_direct(user, throw_datum.starting_turf, round(target.throw_range * 1.5, 1), offset = angle + (rand(-1, 1) * 15))
+	throw_datum.finalize(hit = FALSE)
+	target.mouse_opacity = MOUSE_OPACITY_TRANSPARENT //dont mess with our ball
+	target.color = list(1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,3) //make them super light
+	animate(target, 0.5 SECONDS, color = null, flags = ANIMATION_PARALLEL)
+	user.color = list(1,0,0,0, 0,1,0,0, 0,0,1,0, 0,0,0,3)
+	animate(user, 0.5 SECONDS, color = null, flags = ANIMATION_PARALLEL)
+	playsound(src, 'sound/items/baseballhit.ogg', 100, TRUE)
+	user.do_attack_animation(target, used_item = src)
+	ADD_TRAIT(user, TRAIT_IMMOBILIZED, type)
+	addtimer(CALLBACK(src, .proc/launch_back, target, user, return_to_sender, datum_throw_speed), 0.5 SECONDS)
+	return TRUE
+
+/obj/item/melee/baseball_bat/proc/launch_back(atom/movable/target, mob/living/user, turf/target_turf, datum_throw_speed)
+	playsound(target, 'sound/magic/tail_swing.ogg', 50, TRUE)
+	REMOVE_TRAIT(user, TRAIT_IMMOBILIZED, type)
+	target.mouse_opacity = initial(target.mouse_opacity)
+	target.add_filter("baseball_launch", 3, motion_blur_filter(1, 3))
+	target.throwforce *= 2
+	target.throw_at(target_turf, get_dist(target, target_turf), datum_throw_speed + 1, user, callback = CALLBACK(src, .proc/on_hit, target))
```

Comment:
will on_hit() always get called? even if, say, the baseball bat gets deleted or the ball never ends up hitting anything? in other words, are we 10,000% sure that people can't use this to give an object an arbitrarily high throwforce?